Update user's guide

lucasverleyen opened the pull request:
Fixes #1444 and #1408

annadellisola reviewed (changes requested):
Thank you @lucasverleyen for addressing the issue. It looks good! I have made mainly some textual suggestions on the ReleaseNotes

lucasverleyen commented:
!! I'll add the logo to the user's guide as well, once #1452 is merged.

annadellisola commented on README.md:
```diff
@@ -143,3 +77,21 @@ doi={10.1080/19401493.2018.1428361},
+[References](https://github.com/open-ideas/IDEAS/blob/master/IDEAS/UsersGuide/References.mo) section of the
+[User's Guide](https://github.com/open-ideas/IDEAS/tree/master/IDEAS/UsersGuide).
+
+## Applications of IDEAS
```
I also thought it would be nice to add a similar plot to have a visual impact:
<img width="1416" height="310" alt="image" src="https://github.com/user-attachments/assets/c0fa741d-aeec-4607-bdab-4c8daa85105e" />

1. https://www.scopus.com/pages/citationOverview?key=e648aa14-cc43-49c1-ab1a-dc8ad78365db&origin=resultslist
2. https://www.scopus.com/term/analyzer.uri?sort=plf-f&src=s&sid=2120985a9b6f4bbd74c2501f87720e65&sot=a&sdt=a&sl=26&s=REFEID%282-s2.0-85042134194%29&origin=resultslist&count=10&analyzeResults=Analyze+results)

However, this picture should be updated time to time...

lucasverleyen replied:
@annadellisola, as a first try, I used your image and added it to the README. It looks nice, so I would do that :-)

jelgerjansen commented on README.md:
```diff
@@ -1,135 +1,69 @@
-============
-Modelica model environment for Integrated District Energy Assessment Simulations (IDEAS), allowing simultaneous transient simulation of thermal and electrical systems at both building and feeder level. This Modelica library was originally developed by KU Leuven and [3E](https://3e.eu) and is currently developed and maintained by the [Thermal Systems Simulation (The SySi)](http://the.sysi.be) rese ...
+<div align="center">
+<img src="IDEAS/Resources/Images/IDEAS-logo.png" height="115"> <b>v3.0.0</b>
```
The logo can be made bigger in my opinion, as well as the version number

lucasverleyen replied:
Ok! I can increase the image size, but I don't find a way to increase the font size of v3.0.0. A solution could be to include the version number in the image, but this creates a new logo file...

lucasverleyen replied:
Ok! I wanted to show the current version number clearly, but I'll make it a bit clearer in the release history section.

jelgerjansen commented on README.md:
```diff
@@ -1,135 +1,69 @@
 
 ## Tool support
-Our goal is to provide support for Dymola and OpenModelica. Consequently, any tool that supports the full Modelica specification should be able to run our models. Feel free to file a bug report in case we do not adhere to the Modelica specification.
+IDEAS is fully compliant with the [Modelica Specification](https://specification.modelica.org/), and is therefore, in
```
"fully complaint" is a dangerous statement. We test our developments in Dymola, which sometimes allows deviations from the Modelica Standard (see e.g. https://github.com/open-ideas/IDEAS/issues/1317, which is an incompatibility that we solved in the past). "Largely compliant" seems more correct to me.

jelgerjansen commented on README.md:
```diff
@@ -1,135 +1,69 @@
-## Unit tests
-The library is unit tested using BuildingsPy. Automated unit tests are run on GitHub Actions with a self-hosted runner at KU Leuven.
+## Getting started
+Two tutorials are provided at <code>IDEAS.Examples.Tutorial</code>, serving as demonstrations cases of how the IDEAS library
```
are provided in

jelgerjansen commented on README.md:
```diff
@@ -1,135 +1,69 @@
+can be used.
+1. <code>IDEAS.Examples.Tutorial.SimpleHouse</code> contains examples with step-by-step instructions for how to build a system
+model for a simple house with a heating system, ventilation, and weather boundary conditions.
+2. <code>IDEAS.Examples.Tutorial.DetailedHouse</code> ontains examples with step-by-step instructions for how to build a system
```
contains

jelgerjansen commented on README.md:
```diff
@@ -1,135 +1,69 @@
+Two tutorials are provided at <code>IDEAS.Examples.Tutorial</code>, serving as demonstrations cases of how the IDEAS library
+can be used.
+1. <code>IDEAS.Examples.Tutorial.SimpleHouse</code> contains examples with step-by-step instructions for how to build a system
+model for a simple house with a heating system, ventilation, and weather boundary conditions.
```
Maybe emphasize that this one uses a very simple building model, not using the IDEAS building envelope component models.

jelgerjansen commented on README.md:
```diff
@@ -143,3 +77,21 @@ doi={10.1080/19401493.2018.1428361},
+## Release history
+IDEAS v3.0.0 was released on May 3, 2022. This release includes an update to the Modelica Standard Library MSL 4.0.0.
+Detailed release notes are available within the library under the following link:
+[ReleaseNotes](https://github.com/open-ideas/IDEAS/tree/master/IDEAS/UsersGuide/ReleaseNotes).
```
Fix URL (file not found)

lucasverleyen replied:
This link will work when this PR is merged. Now, this PR is still on my branch, but once merged, it will be part of the master branch. If I now refer to my branch, the link will break after merging...

jelgerjansen commented on README.md:
```diff
@@ -143,3 +77,21 @@ doi={10.1080/19401493.2018.1428361},
+[ReleaseNotes](https://github.com/open-ideas/IDEAS/tree/master/IDEAS/UsersGuide/ReleaseNotes).
+Notes for the previous releases notes can also be found there. 
+
+## Documentation
```
I think it is worth to mention the development papers of IDEAS, as this allows user to more easily find the relevant background for different models (e.g. the one-port interzonal airflow implementation). Some of these development papers might be mentioned in the documentation section of the models, but I'm not sure if this was done consistently.
@lucasverleyen, I let you decide if you prefer to add these or not ;)

lucasverleyen replied:
see comment below

jelgerjansen commented on README.md:
```diff
@@ -143,3 +77,21 @@ doi={10.1080/19401493.2018.1428361},
+## Documentation
+The IDEAS library offers comprehensive documentation. Each model includes a dedicated documentation section explaining its implementation details.
+Many models also reference relevant scientific papers. An overview of the key reference papers is provided in the
+[References](https://github.com/open-ideas/IDEAS/blob/master/IDEAS/UsersGuide/References.mo) section of the
```
This one refers to papers that when developing the IDEAS library, not sure if this is important to mention in the README These papers where used to develop the core models of IDEAS (eg. convection coefficient correlations), which are not of interest for general users.

lucasverleyen replied:
See https://github.com/open-ideas/IDEAS/pull/1445#issuecomment-3151647660

jelgerjansen reviewed (changes requested):
Thanks for the huge effort Lucas!
I already fixed some typo/lay-out stuff myself and left a few comments regarding the content.

lucasverleyen commented:
> @jelgerjansen @annadellisola I have restructured and updated the User's Guide in the main IDEAS package. Could you please review this? For now, only Filip's funding was mentioned. I partially completed the list in IDEAS/UsersGuide/Acknowledgements. I still need to check with Lieve for the KU Leuven part. @kldjonge, do you want to mention your/UGent funding for IDEAS developments?
> 
> Before the new IDEAS release, we need to complete IDEAS/UsersGuide/ReleaseNotes/Version_4_0_0.

@jelgerjansen I have double-checked the list of funding and contributors with Lieve.

jelgerjansen commented:
@lucasverleyen OK for me to restructure the References.mo file.
I swept over the .mo files and could find all papers mentioned in the references, except the work of Jürges, Kenelly, and Waldon.

lucasverleyen commented:
@jelgerjansen @annadellisola I have restructured the References.mo file.
- I have added the correct hyperlinks of the "main references" to the respective models, and I have removed those from the References.mo.
- I have added the papers on the development of IDEAS, which were before in the readme of the repo. However: i) I did not include the paper on StROBe (has nothing to do with IDEAS development), ii) I did not add the PhD of Ruben Baetens (the papers are already included), iii) I added two more papers from the applications, which are more on the development in my opinion.

- I could not find Kenelly. But when I looked for "triangle", I was referred to IDEAS.Electrical.AC.ThreePhaseBalanced.Loads.Impedance. There, the parameter "star" is declared to define a star or triangle configuration. I have added Kenelly (a basic paper about star and triangle configurations in electrical networks) there.
- I could find "Waldon" using the title, but it is Walton, so this problem is solved.

- I could not find ISO 13370 (on the heat load calculations). @jelgerjansen, could you find it? Where do I add this reference?

- I could not find Defraye, also not when I looked for the keywords in the title. I suggest removing the reference.
- I could not find Jürges, also not when I looked for the keywords (both in German and English). I suggest removing the reference. It seems like a basic reference.

Can you do a new (final?) review?

annadellisola commented on IDEAS/Buildings/Components/BaseClasses/RadiativeHeatTransfer/ZoneLwDistribution.mo:
```diff
@@ -192,8 +192,10 @@ It may be used when the initial equation that computes the view factors does not
-Carroll, J.A. 1980. An \"MRT method\" of computing radiant energy exchange in rooms. Proceedings of the 2nd Systems Simulation and Economics Analysis Conference, January 23-25
+[Liesen 1997]: R.J. Liesen, and C.O. Pedersen, 
+\"<a href=\"https://iifiir.org/en/fridoc/an-evaluation-of-inside-surface-heat-balance-models-for-cooling-load-16626\">An evaluation of inside surface heat balance models for cooling load calculations</a>,\" 
+<i>ASHRAE Transactions</i>, vol. 103, no. 2, pp. 485-502, 1997.
```
```suggestion
<i>ASHRAE Transactions</i>, vol. 103, no. 2, pp. 485-502, 1997.<br>

annadellisola reviewed (commented):
Thanks a lot @lucasverleyen for all the effort! I gave some minor comments on the references. 
Regarding the paper you did not find:
- ISO 13370: could be referenced in IDEAS.Buildings.Components.SlabOnGround, where it seems that UEqui is calculated according to the standard
- Defraye 2011: I was also not able to find a direct reference. IDEAS.Buildings.Components.OuterWall might use some correlations mentioned in the paper, but I did not manage to find the exact one
- Jurges 1924: no reference found

lucasverleyen commented on IDEAS/UsersGuide/References.mo:
```diff
@@ -1,65 +1,99 @@
+<li>
+F. Jorissen, W. Boydens, L. Helsen,
+\"<a href=\"https://doi.org/10.1080/19401493.2018.1544277\">Model implementation and verification of the envelope, HVAC and controller of an office building in Modelica</a>\",
+<i>Journal of Building Performance Simulation</i>, vol. 12, no. 4, pp. 445&ndash;464, 2019.
```
It was added before... I thought it was maybe related to an example in IDEAS, but I didn't verify. We can remove it.

jelgerjansen replied:
I would remove this, this is rather an application. Instead, Filip's PhD can be included, as that one also described the IDEAS library in more detail.

lucasverleyen replied:
Done

lucasverleyen commented on IDEAS/UsersGuide/References.mo:
How do you mean? What is the difference, and to what do you compare?

lucasverleyen replied:
Ah, I see what you mean. I didn't see any added value in repeating [name, year] and structuring it in a table. (+ it was more work to create the table than bullet points, so that's the reason haha ;-) )

jelgerjansen commented on IDEAS/Buildings/Components/BaseClasses/ConvectiveHeatTransfer/InteriorConvection.mo:
```diff
@@ -116,10 +116,16 @@ a thermal circuit formulation for the direct radiant exchange between surfaces c
 <p>
-[Awbi 1999]: H.B. Awbi, A. Hatton, \"Natural convection from heated room surfaces\", <i>Energy and Buildings</i>, vol. 30, no. 3, pp. 233&ndash;244, 1999.<br>
+[Awbi 1999]: H.B. Awbi, and A. Hatton, 
+\"<a href=\"http://www.sciencedirect.com/science/article/pii/S0378778899000043\">Natural convection from heated room surfaces</a>\", 
```
Maybe refer to the doi hyperlink if possible? (https://doi.org/10.1016/S0378-7788(99)00004-3)
I don't think the current hyperlink will change, but the doi one seems more "future proof".

lucasverleyen replied:
Good catch! I checked all links and replaced them with a DOI hyperlink where possible.

jelgerjansen commented on IDEAS/Buildings/Components/BaseClasses/RadiativeHeatTransfer/SwWindowResponse.mo:
```diff
@@ -137,8 +137,14 @@ equation
+    Icon(graphics={  Rectangle(fillColor = {192, 192, 192}, pattern = LinePattern.None, fillPattern = FillPattern.Backward, extent = {{-80, 90}, {80, 70}}), Line(points = {{-80, 70}, {80, 70}}, pattern = LinePattern.None, thickness = 0.5), Line(points = {{44, 40}, {44, -50}}, color = {127, 0, 0}), Line(points = {{44, 40}, {38, 30}}, color = {127, 0, 0}), Line(points = {{44, 40}, {50, 30}}, color  ...
+    Documentation(info="<html>
+<p>
+The properties for absorption by and transmission through the glazing are taken into account depending on the angle of incidence of solar irradiation and are based on the output of the [WINDOW 6.3] software, i.e. the shortwave properties itselves based on the layers in the window are not calculated in the model but are input parameters.
```
Exactly version 6.3? Because some data records refer to v7.3.4.0.
Maybe we can use a more general reference for the WINDOW software?

jelgerjansen commented on IDEAS/Electrical/AC/ThreePhasesBalanced/Loads/Impedance.mo:
```diff
@@ -141,6 +141,12 @@ When the connection type changes from
 is recomputed in such a way that the nominal power consumed by the impedance
 does not change.
 </p>
+<h4>References</h4>
+<p>
+[Kenelly 1899]: A.E. Kenelly, 
+\"Equivalence of triangles and stars in conducting networks\", 
+<i>Electrical World and Engineer</i>, vol. 34, pp. 413&ndash;414, 1899.
+</p>
```
This is an IBPSA model, so it shouldn't be updated.
I would remove this reference overall (it's "common" knowledge in engineering).

jelgerjansen commented on IDEAS/UsersGuide/References.mo:
```diff
@@ -1,65 +1,99 @@
+<i>Journal of Building Performance Simulation</i>, vol. 11, no. 6, pp. 639&ndash;654, 2018.
+</li>
+<li>
+B. van der Heijde, M. Fuchs,  C. Ribas Tugores, G. Schweiger, K. Sartor, D. Basciotti, D. Müller, C. Nytsch-Geusen, M. Wetter, and L. Helsen,
+\"<a href=\"https://doi.org/10.1016/j.enconman.2017.08.072\">Dynamic equation-based thermo-hydraulic pipe model for district heating and cooling systems</a>\",
+<i>Energy Conversion and Management</i>, vol. 151, pp. 158&ndash;169; 2017.
```
FYI: This model is in the IBPSA library, but can be kept here as Bram was at KU Leuven at that time.

lucasverleyen commented:
> Thanks a lot @lucasverleyen for all the effort! I gave some minor comments on the references. Regarding the papers you did not find:
> 
> * ISO 13370: could be referenced in IDEAS.Buildings.Components.SlabOnGround, where it seems that UEqui is calculated according to the standard
> * Defraye 2011: I was also not able to find a direct reference. IDEAS.Buildings.Components.OuterWall might use some correlations mentioned in the paper, but I did not manage to find the exact one
> * Jurges 1924: no reference found

@annadellisola thanks for your suggestions. I have added ISO 13370 as explained. I have removed Defraye and Jurges.

lucasverleyen commented:
@annadellisola @jelgerjansen thanks again for your useful input! I have addressed all comments. I think this PR is ready.
